src/GraphQL/CacheableGraphqlValue.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * @file
+ * Wrapper class for values resolved through GraphQL resolvers, which includes
+ * Drupal cache metadata.
+ */
+
+namespace Drupal\graphql\GraphQL;
+
+use Drupal\Core\Cache\CacheableDependencyInterface;
+use Drupal\Core\Cache\CacheableMetadata;
+
+class CacheableGraphqlValue extends CacheableMetadata {
```

Comment:
Let's call this CacheableLeafValue. It should only be used for leaf values and that makes it clearer.